dali/operators/generic/erase/erase.cu

```diff
@@ -0,0 +1,163 @@
+as_boxes(TensorListView<Storage, int32_t, 3> tlv) {
+  TensorListShape<1> new_shape(tlv.shape.size());
+  for (int i = 0; i < tlv.shape.size(); ++i)
+    new_shape.set_tensor_shape(i, {tlv.shape[i][0]});
+  using ptrs_t = kernels::ibox<ndim>* const*;
+  return {reinterpret_cast<ptrs_t>(tlv.data.data()), new_shape};
+}
+
+template <int ndim, typename Storage>
+TensorListView<Storage, const kernels::ibox<ndim>, 1>
+as_boxes(TensorListView<Storage, const int32_t, 3> tlv) {
+  TensorListShape<1> new_shape(tlv.shape.size());
+  for (int i = 0; i < tlv.shape.size(); ++i)
+    new_shape.set_tensor_shape(i, {tlv.shape[i][0]});
+  using ptrs_t = const kernels::ibox<ndim>* const*;
+  return {reinterpret_cast<ptrs_t>(tlv.data.data()), new_shape};
+}
+
+}  // namespace detail
+
+template <typename T, int Dims, int channel_dim>
+class EraseImplGpu : public OpImplBase<GPUBackend> {
+ public:
+  using EraseKernel = kernels::EraseGpu<T, Dims, channel_dim>;
+
+  explicit EraseImplGpu(const OpSpec &spec)
+  : spec_(spec)
+  , batch_size_(spec.GetArgument<int>("batch_size")) {
+      kmgr_.Resize<EraseKernel>(1, 1);
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc,
+                 const workspace_t<GPUBackend> &ws) override {
+    const auto &input = ws.template InputRef<GPUBackend>(0);
+    auto layout = input.GetLayout();
+    auto type = input.type();
+    auto shape = input.shape();
+    AcquireArgs(ws, shape, layout);
+
+    auto in_view = view<const T, Dims>(input);
+    kmgr_.Initialize<EraseKernel>();
+    ctx_.gpu.stream = ws.stream();
+    auto regions_view = detail::as_boxes<Dims>(view<const int32_t, 3>(regions_gpu_));
+    kmgr_.Setup<EraseKernel>(0, ctx_, in_view, regions_view, make_cspan(fill_values_));
+
+    output_desc.resize(1);
+    output_desc[0] = {shape, input.type()};
+    return true;
+  };
+
+  void RunImpl(workspace_t<GPUBackend> &ws) override {
+    auto input = view<const T, Dims>(ws.template InputRef<GPUBackend>(0));
+    auto output = view<T, Dims>(ws.template OutputRef<GPUBackend>(0));
+    auto regions_view = detail::as_boxes<Dims>(view<const int32_t, 3>(regions_gpu_));
+    kmgr_.Run<EraseKernel>(0, 0, ctx_, output, input, regions_view, make_cspan(fill_values_));
+  }
+
+  void AcquireArgs(const DeviceWorkspace &ws, TensorListShape<> in_shape,
+                   TensorLayout in_layout) {
+    fill_values_ = spec_.template GetRepeatedArgument<float>("fill_value");
+    auto args = detail::GetEraseArgs<T, Dims>(spec_, ws, in_shape, in_layout);
+    auto regions_shape = TensorListShape<3>(batch_size_);
+    for (int i = 0; i < batch_size_; ++i) {
+      auto n_regions = static_cast<int>(args[i].rois.size());
+      regions_shape.set_tensor_shape(i, {n_regions, 2, Dims});
+    }
+    TensorList<CPUBackend> regions_cpu;
+    regions_cpu.set_type(TypeTable::GetTypeInfo(TypeTable::GetTypeID<int32_t>()));
+    regions_cpu.Resize(regions_shape);
+    auto regions_tlv = detail::as_boxes<Dims>(view<int32_t, 3>(regions_cpu));
```

Comment:
And you don't need this `as_boxes` thing at all, you can take a `view<ibox<Dims>>(regions_cpu)` and place the data in it directly. AFAIK it should work.

Reply:
done